.github/workflows/main.yml

```diff
@@ -0,0 +1,54 @@
+name: Unittests & Auto-publish
+
+# Allow to trigger the workflow manually (e.g. when deps changes)
+on: [push, workflow_dispatch]
+
+jobs:
+  unittest-job:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}-${{ matrix.python-version }}
+      cancel-in-progress: true
+
+    steps:
+    - uses: actions/checkout@v3
+
+    # Install deps
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - run: pip --version
+    - run: pip install -e .
+    - run: pip freeze
+
+    # Run tests
+    - name: Run core tests
+      run: python -m unittest
```

Comment:
We can try to use https://github.com/google/googletest

Reply:
Using `absl.testing.absltest`. `googletest` is the C++ testing framework.